File: src/client/java/net/oxyopia/vice/config/ModMenuIntegration.java
package net.oxyopia.vice.config;

import com.terraformersmc.modmenu.api.ModMenuApi;
import com.terraformersmc.modmenu.api.ConfigScreenFactory;

public class ModMenuIntegration implements ModMenuApi {
	@Override
	public ConfigScreenFactory<?> getModConfigScreenFactory() {
		return parent -> new Config().gui();
	}
}

File: src/main/java/net/oxyopia/vice/config/ModMenuIntegration.kt
package net.oxyopia.vice.config

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import net.minecraft.client.gui.screen.Screen
import net.oxyopia.vice.Vice

class ModMenuIntegration : ModMenuApi {
	override fun getModConfigScreenFactory(): ConfigScreenFactory<*> = ConfigScreenFactory<Screen?> { Vice.config.gui() }
}